src/librustc_resolve/macros.rs

```diff
@@ -268,7 +268,7 @@ impl<'a> base::Resolver for Resolver<'a> {
                                 if k > 0 {
                                     tokens.push(TokenTree::Token(path.span, Token::ModSep).into());
                                 }
-                                let tok = Token::Ident(segment.identifier);
+                                let tok = Token::from_ast_ident(segment.identifier);
```

Comment:
Same as https://github.com/rust-lang/rust/pull/48942/files#r175613878

Reply:
I changed this in `from_ast_ident`, since it seems this is the best behavior in all these cases.